spec.html

```diff
@@ -21598,16 +21637,42 @@
           </emu-note>
         </emu-clause>
 
+        <emu-clause id="sec-resolutionfailure" aoid="ResolutionFailure">
+          <h1>Runtime Semantics: ResolutionFailure( _resolveMode_ )</h1>
+          <p>The ResolutionFailure abstract operation performs the following steps:</p>
+          <emu-alg>
+            1. If _resolveMode_ is `"namespace"` or `"star"`: return *undefined*.
+            1. Assert: _resolveMode_ is `"normal"`.
+            1. Throw a new *SyntaxError* exception.
+          </emu-alg>
+        </emu-clause>
+
+        <emu-clause id="sec-propagateresolution" aoid="PropagateResolution">
+          <h1>Runtime Semantics: PropagateResolution( _module_, _resolution_ )</h1>
+          <p>The PropagateResolution abstract operation performs the following steps:</p>
+          <emu-alg>
+            1. If _resolution_ is an abrupt completion, then
+              1. If _module_.[[Status]] is `"errored"`, then
+                1. Assert: _module_.[[ErrorCompletion]] is _resolution_.
```

Comment:
This is a completion value though, so SameValue doesn't quite apply. Cf. "Assert: module.[[Realm]] is not undefined."